web/pdf_page_view.js

```diff
@@ -541,26 +547,30 @@ class PDFPageView {
       }
     );
 
-    if (this.annotationLayerFactory) {
-      if (!this.annotationLayer) {
-        this.annotationLayer = this.annotationLayerFactory.createAnnotationLayerBuilder(
-          div,
-          pdfPage,
-          this._annotationStorage,
-          this.imageResourcesPath,
-          this.renderInteractiveForms,
-          this.l10n
-        );
+    return this.hasJSActionsPromise.then(hasJSActions => {
+      if (this.annotationLayerFactory) {
+        if (!this.annotationLayer) {
+          this.annotationLayer = this.annotationLayerFactory.createAnnotationLayerBuilder(
+            div,
+            pdfPage,
+            this._annotationStorage,
+            this.imageResourcesPath,
+            this.renderInteractiveForms,
+            this.l10n,
+            this.enableScripting,
+            hasJSActions
+          );
+        }
+        this._renderAnnotationLayer();
       }
-      this._renderAnnotationLayer();
-    }
-    div.setAttribute("data-loaded", true);
+      div.setAttribute("data-loaded", true);
 
-    this.eventBus.dispatch("pagerender", {
-      source: this,
-      pageNumber: this.id,
+      this.eventBus.dispatch("pagerender", {
+        source: this,
+        pageNumber: this.id,
+      });
+      return resultPromise;
```

Comment:
You *absolutely* can't delay this event, since that will affect things like clean-up (and thus resource usage) in the viewer negatively.